client/generator/src/main/java/io/openlineage/client/JavaPoetGenerator.java

```diff
@@ -0,0 +1,174 @@
+          fieldNames.add(f.getName());
+          if (f.getName().equals("_schemaURL")) {
+            String schemaURL = baseURL + "#/definitions/" + type.getName();
+            constructor.addCode("this.$N = $S;\n", f.getName(), schemaURL);
+          } else {
+            constructor.addJavadoc("@param $N $N\n", f.getName(), f.getDescription() == null ? "" : f.getDescription());
+            constructor.addParameter(
+                ParameterSpec.builder(getTypeName(f.getType()), f.getName())
+                .addAnnotation(AnnotationSpec.builder(JsonProperty.class).addMember("value", "$S", f.getName()).build())
+                .build());
+            constructor.addCode("this.$N = $N;\n", f.getName(), f.getName());
+          }
+          MethodSpec getter = getter(f)
+              .addModifiers(Modifier.PUBLIC)
+              .addCode("return $N;", f.getName())
+              .build();
+          classBuilder.addMethod(getter);
+        }
+         // additionalFields
+        if (type.isHasAdditionalProperties()) {
+          String fieldName = "additionalProperties";
+          TypeName additionalPropertiesValueType = type.getAdditionalPropertiesType() == null ? ClassName.get(Object.class) : getTypeName(type.getAdditionalPropertiesType());
+          TypeName additionalPropertiesType = ParameterizedTypeName.get(ClassName.get(Map.class), ClassName.get(String.class), additionalPropertiesValueType);
+          classBuilder.addMethod(MethodSpec
+              .methodBuilder("get" + titleCase(fieldName))
+              .addJavadoc("additional properties")
+              .returns(additionalPropertiesType)
+              .addModifiers(Modifier.PUBLIC)
+              .addCode("return $N;", fieldName)
+              .addAnnotation(AnnotationSpec.builder(JsonAnyGetter.class).build())
+              .build());
+          classBuilder.addField(
+              FieldSpec.builder(additionalPropertiesType, fieldName, PUBLIC, FINAL)
+              .addAnnotation(JsonAnySetter.class)
+              .build());
+
+          constructor.addCode(CodeBlock.builder().addStatement("this.$N = new $T<>();\n", fieldName, HashMap.class).build());
+        }
+        classBuilder.addMethod(constructor.build());
+        TypeSpec clss = classBuilder.build();
+        builder.addType(clss);
+      }
+    }
+
+  }
+
+  private Builder getter(Field f) {
+    Builder builder = MethodSpec
+        .methodBuilder("get" + titleCase(f.getName()))
+        .returns(getTypeName(f.getType()));
+    if (f.getDescription() != null) {
+      builder.addJavadoc("$N", f.getDescription());
+    }
+    return builder;
+  }
+
+  public static TypeName getTypeName(Type type) {
+    return type.accept(new TypeVisitor<TypeName>(){
+
+      @Override
+      public TypeName visit(PrimitiveType primitiveType) {
+        if (primitiveType.getName().equals("string")) {
+          return ClassName.get(String.class);
+        }
```

Comment:
Add date-time with a validating class? I personally have mixed feelings about it. It may be nice to be able to programmatically resolve scalar types & additional facets via a compile-time type annotation.

Reply:
I suggest we add the support for the `format` as a follow up.